plugin/forward/forward.go

```diff
@@ -82,6 +82,10 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 			fails = 0
 		}
 
+		if plugin.Done(ctx) {
+			return dns.RcodeSuccess, plugin.Error(f.Name(), ctx.Err())
+		}
+
```

Comment:
Should this go [in here](https://github.com/coredns/coredns/pull/2736/files#diff-b12d06e8aa6117ce05d5715e1c8930c3R115) to break before tcp retry if we need to?